cypress/integration/channel/long_post_attachments_spec.js

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+describe('Test Long post with attachements, Test key= M14322', () => {
+    before(() => {
+        cy.login('user-1');
+        cy.visit('/');
+
+        // Add 4 attachments
+        cy.uploadFile('#fileUploadButton input', '../fixtures/mattermost-icon.png', 'image/png');
+        cy.uploadFile('#fileUploadButton input', '../fixtures/mattermost-icon.png', 'image/png');
+        cy.uploadFile('#fileUploadButton input', '../fixtures/mattermost-icon.png', 'image/png');
+        cy.uploadFile('#fileUploadButton input', '../fixtures/longTextPost.txt', 'text/txt');
+
+        // Add a long post
+        cy.fixture('./longTextPost.txt').then((data) => {
+            cy.postMessage(data);
+        });
+        cy.wait(500);
+    });
+
+    it('Attachment previews/thumbnails display as expected, when viewing full or partial post', () => {
+        // Get the last PostId and check if the 4 attachments are posted
+        cy.getLastPostId().then((postID) => {
+            cy.get(`#${postID}_message>.post-image__columns`).children().should('have.length', '4');
+        });
+    });
+
+    it('Can click one of the attachments and cycle through the multiple attachment previews as usual', () => {
+        // Preview the first attachment
+        cy.getLastPostId().then((postID) => {
+            cy.get(`#${postID}_message>.post-image__columns`).children().first().click();
+        });
+
+        // Preview the next attachments
+        for (var i = 0; i < 3; i++) {
+            cy.get('#previewArrowRight>.image-control').click();
```

Comment:
This can be done as,
```suggestion
            cy.get('#previewArrowRight').click();
```

I think that clicking on `previewArrowRight` is not enough without verifying something on file preview modal. Though not specifically stated, I recommend to check that the file preview is showing correctly by at least checking its label, like the "File 1 of 4".  You may add new ID as [id='popoverBar'](https://github.com/mattermost/mattermost-webapp/blob/14b84decb89c66e82b4b731c722a45e3d1429675/components/view_image/popover_bar/popover_bar.jsx#L95) to easily check the content.

Reply:
Thank you @saturninoabril for your helpful comments!. I made some changes, please check.